controllers/ptpinterface_controller.go

```diff
@@ -606,6 +606,23 @@ func (r *PtpInterfaceReconciler) Reconcile(ctx context.Context, request ctrl.Req
 		return reconcile.Result{}, err
 	}
 
+	// Restore the PtpInterface status
+	if r.checkRestoreInProgress(instance) {
+		r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Restoring '%s' PtpInterface resource status without doing actual reconciliation", instance.Name)
+		if err := r.RestorePtpinterfaceStatus(instance); err != nil {
+			return reconcile.Result{}, err
+		}
+		if err := r.ClearAnnotation(instance); err != nil {
+			return reconcile.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if instance.Status.ObservedGeneration == instance.ObjectMeta.Generation &&
+		instance.Status.Reconciled == true {
+		return ctrl.Result{}, nil
+	}
```

Comment:
Just keep it as `instance.Status.Reconciled` instead of `instance.Stauts.Reconciled == true`.
This will also fix golang-lint issues.

Reply:
Changes are done